src/code/z_fbdemo_circle.c

```diff
@@ -1,19 +1,134 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Start.s")
+Gfx D_801D0D00[] = { 0xE7000000, 0x00000000, 0xEF002CF0, 0x0342524D, 0xFA0000FF,
+                     0x00000001, 0xF9000000, 0x00000001, 0xDF000000, 0x00000000 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Init.s")
+// @bug TransitionCircle_Update may need to be casted? (void (*)(void* transition, s32 updateRate))?
+const TransitionInit TransitionCircle_InitVars = {
+    TransitionCircle_Init,   TransitionCircle_Destroy, TransitionCircle_Update,   TransitionCircle_Draw,
+    TransitionCircle_Start,  TransitionCircle_SetType, TransitionCircle_SetColor, NULL,
+    TransitionCircle_IsDone,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Destroy.s")
+void TransitionCircle_Start(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Update.s")
+    this->stepValue = 0.1f;
+    if (this->direction == 0) {
+        this->unk_10 = 0.0f;
+        this->unk_0C = 1.0f;
+    } else {
+        this->unk_0C = 0.0f;
+        this->unk_10 = 1.0f;
+    }
+    this->unk_04 = this->unk_0C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_SetColor.s")
+void* TransitionCircle_Init(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_SetType.s")
+    bzero(this, sizeof(TransitionCircle));
+    this->unk_15 = 1;
+    this->texture = D_801DE890;
```

Comment:
I think this was commented in discord, but this is the lens texture in OoT, which seems to be used as a more generic circle texture in MM. It's currently called `gLensOfTruthMaskTex`, but I'm not sure what the best name is. Anyone else have thoughts?

Reply:
Perhaps just `gCircleTex`? Or `gCircleMaskTex` if it is always used as a mask?

Reply:
I went with gCircleTex, but if there are other thoughts, I'm okay to change

Reply:
`gCircleTex` is fine for now, but I would say it would need another look later, especially since there are many circle textures/masks out there presumably (shadows for one)

Reply:
I'll sync this up with my magic PR